### PR TITLE
Edit: Fix incorrect prompt used for enterprise chat models

### PR DESCRIPTION
**Pull request by umpox**
## Description

Fixes an issue where an enterprise instance would have `chatModel` set to something non-claude specific.

We would still override the model, which resulted in weird bugs

Also improves the type safety again, as this seemed to have regressed

## Test plan

1. Connect to an enterprise instance with the `chatModel` set to an OpenAI model
2. Run an edit command
3. Check that it works as expected

### CODE REVIEW

File: lib/shared/src/models/dotcom.ts [hunk 1]
```diff
@@ -148,7 +148,7 @@ const DEFAULT_DOT_COM_MODELS: ModelProvider[] = [
         contextWindow: { input: CHAT_INPUT_TOKEN_BUDGET, output: CHAT_OUTPUT_TOKEN_BUDGET },
         deprecated: true,
     },
-]
+] as const satisfies ModelProvider[]
```
Comment (umpox) [inline lib/shared/src/models/dotcom.ts:151]: We rely on this for type safety when determining the `EditModel` type, but it was accidentally removed on an unrelated PR.

I'll see if there's any way to detect if this breaks 🤔

File: lib/shared/src/models/types.ts [hunk 2]
```diff
@@ -38,6 +50,18 @@ export type ChatModel =
       }[keyof Models]['model']
     | (string & {})
 
+/**
+ * Available models for Chat.
```
Comment (umpox) [inline lib/shared/src/models/types.ts:54]:
```suggestion
 * Available providers for Chat.
```

**Review by abeatrix — APPROVED** [review]
Looks like it was my doing, my bad! Thanks for fixing it

**Comment by abeatrix** [comment]
> Fixes an issue where an enterprise instance would have chatModel set to something non-claude specific.

What is enterprise instance is using non-Claude model? Or are you talking about something else?

**Comment by umpox** [comment]
@abeatrix If the enterprise instance has a chat model set to like `openai/gpt4` or whatever the model name is. We were falling back to the claude prompt if we didn't know 100% what it was, so it was erroring for those model strings

File: vscode/src/edit/utils/edit-models.ts [hunk 3]
```diff
@@ -5,7 +5,17 @@ export function getEditModelsForUser(authStatus: AuthStatus): ModelProvider[] {
     return ModelProvider.getProviders(ModelUsage.Edit, !authStatus.userCanUpgrade)
 }
 
-export function getOverridenModelForIntent(intent: EditIntent, currentModel: EditModel): EditModel {
+export function getOverridenModelForIntent(
+    intent: EditIntent,
+    currentModel: EditModel,
+    authStatus: AuthStatus
+): EditModel {
+    if (!authStatus.isDotCom) {
```
Comment (kalanchan) [inline vscode/src/edit/utils/edit-models.ts:13]: nice 😎

**Review by kalanchan — APPROVED** [review]
looks good, thanks for fixing this!

@umpox could you merge as soon as you can? I'd like to have this in the latest agent commit for the next JB release

**Comment by umpox** [comment]
@kalanchan Enabled auto-merge so just need CI to pass

**Pull request merged**